### DISCUSSION TITLE
Jitsi and keycloak

### DISCUSSION THREAD

**Discussion by EzechielBrou** [keycloak/keycloak#35686] [Adapter deprecation]
Hi all, I encounter a problem when binding keycloak and jitsi-meet. I want to permit jitsi to authenticate his Users via keycloak which it’s sync with LDAP server. So I have already used nordeck/keycloak-jitsi-adapter, when I want to join a session after her creation, I receive a message which tell me if I’m a host, I click on host, I’m redirected on keycloak authentication, when I finish, I join the session for few seconds, and I receive yet the same message of host and it’s the same thing until disconnection.

**Comment by codespearhead** [comment]
Jitsi doesn't appear to support SSO out of the box [01], which is why I believe you're using [02], a third-party OIDC client library for Jitsi.

It seems your access token is expiring shortly after being issued. This could be due to a clock skew between your machine and the Keycloak server, or because the access token lifespan in Keycloak is misconfigured to just a few seconds [03].

If neither resolves the issue, and since that OIDC client adapter [02] is not officially part of the Keycloak project, could you create an issue in their repository? Additionally, consider marking this comment as the accepted answer [04] for this discussion to prevent it from showing as unanswered in the GitHub discussions tab.

By the way, that issue [01] references a similar, more popular project that includes an OIDC client adapter for Jitsi [05] (they call it "SSO setup" in their README).

***

[01] https://github.com/jitsi/jitsi-meet/issues/7046
[02] https://github.com/nordeck/jitsi-keycloak-adapter
[03] https://blog.elest.io/keycloak-token-management-expiration-revocation-and-renewal
[04] https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer
[05] https://github.com/H2-invent/jitsi-admin
